src/test/java/seedu/address/logic/commands/sortcommands/SortCompanyCommandTest.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands.sortcommands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalCompanies.getTypicalAddressBook;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.commandresults.RegularCommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.company.internship.InternshipInterviewDateTime;
+
+public class SortCompanyCommandTest {
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    }
+    @Test
+    public void execute_sortCompany_success() throws Exception {
+        SortCompanyCommand sortCompanyCommand = new SortCompanyCommand(
+            Optional.of(new InternshipInterviewDateTime(LocalDateTime.of(2023, 12, 1, 0, 0))),
+            Optional.of(new InternshipInterviewDateTime(LocalDateTime.of(2024, 1, 20, 0, 0)))
+        );
+
+        assertEquals(new RegularCommandResult(SortCompanyCommand.MESSAGE_SUCCESS), sortCompanyCommand.execute(model));
+
+        // Here you might want to add assertions to check that the companies in the model are sorted correctly.
```

Comment:
Is this for when we work on bug fixing in 1.4? Very thorough of you to add this comment as it would be very helpful for test case writing later on!

Reply:
Currently, I do not have any integration tests. Hopefully, this comment will remind me of this.